src/background.js

```diff
@@ -184,6 +184,12 @@ const TRANSFER_TX_TYPE = 'transfer';
         }
         break;
       case RECEIVE_APPROVAL_REQUEST:
+        const existingRequestIndex = approvalRequests.findIndex(r => r.type === msg.requestType && r.from === msg.from);
+        if (existingRequestIndex >= 0) {
+          approvalRequests.splice(existingRequestIndex, 1);
+          approvalRequestsCallbacks.splice(existingRequestIndex, 1);
+        }
+
```

Comment:
This code checks if there is an existing approval request with the same type and origin and if true it deletes the old request and saves the new one.

Reply:
Let's add this in the code itself.